gobblin-iceberg/src/test/java/org/apache/gobblin/iceberg/writer/GobblinMCEWriterTest.java

```diff
@@ -0,0 +1,166 @@
+ */
+@PrepareForTest({GobblinConstructorUtils.class, FileSystem.class})
+@PowerMockIgnore("javax.management.*")
+public class GobblinMCEWriterTest extends PowerMockTestCase {
+
+  private String dbName = "hivedb";
+  private String tableName = "testTable";
+  private GobblinMCEWriter gobblinMCEWriter;
+  private KafkaStreamingExtractor.KafkaWatermark watermark;
+
+  private GobblinMetadataChangeEvent.Builder gmceBuilder;
+
+  @Mock
+  private MetadataWriter mockWriter;
+
+  @Mock
+  private FileSystem fs;
+
+  @Mock
+  private HiveSpec mockHiveSpec;
+
+  @AfterMethod
+  public void clean() throws Exception {
+    gobblinMCEWriter.close();
+  }
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    gmceBuilder = GobblinMetadataChangeEvent.newBuilder()
+        .setDatasetIdentifier(DatasetIdentifier.newBuilder()
+            .setDataPlatformUrn("urn:li:dataPlatform:hdfs")
+            .setNativeName("/data/tracking/testTable")
+            .build())
+        .setFlowId("testFlow")
+        .setSchemaSource(SchemaSource.EVENT)
+        .setOperationType(OperationType.add_files)
+        .setCluster(ClustersNames.getInstance().getClusterName());
+    watermark = new KafkaStreamingExtractor.KafkaWatermark(
+        new KafkaPartition.Builder().withTopicName("GobblinMetadataChangeEvent_test").withId(1).build(),
+        new LongWatermark(10L));
+
+    State state = new State();
+    state.setProp("gmce.metadata.writer.classes", mockWriter.getClass().getName());
+    Mockito.doNothing().when(mockWriter)
+        .writeEnvelope(
+            Mockito.any(RecordEnvelope.class), Mockito.anyMap(), Mockito.anyMap(), Mockito.any(HiveSpec.class));
+
+    PowerMockito.mockStatic(GobblinConstructorUtils.class);
+    when(GobblinConstructorUtils.invokeConstructor(
+            eq(MetadataWriter.class), eq(mockWriter.getClass().getName()), any(State.class)))
+        .thenReturn(mockWriter);
+
+    PowerMockito.mockStatic(FileSystem.class);
+    when(FileSystem.get(any()))
+        .thenReturn(fs);
+
+    HiveTable mockTable = Mockito.mock(HiveTable.class);
+    when(mockTable.getDbName()).thenReturn(dbName);
+    when(mockTable.getTableName()).thenReturn(tableName);
+
+    mockHiveSpec = Mockito.mock(HiveSpec.class);
+    when(mockHiveSpec.getTable()).thenReturn(mockTable);
+
+    gobblinMCEWriter = new GobblinMCEWriter(new GobblinMCEWriterBuilder(), state);
+  }
+
+  @Test
+  public void testWriteNoOpIfNotSpecified() throws IOException {
+    gmceBuilder.setAllowedMetadataWriters(Arrays.asList("Some other metadata writer"));
+    gobblinMCEWriter.write(
+        new RecordEnvelope<>(gmceBuilder.build(), watermark),
+        new ConcurrentHashMap(), new ConcurrentHashMap(), mockHiveSpec);
+
+    Mockito.verify(mockWriter, never()).writeEnvelope(any(), any(), any(), any());
+  }
+
+  @Test(dataProvider = "AllowMockMetadataWriter")
```

Comment:
We expect calls to the mock writer if the class name is specified in the GMCE or there are no metadata writers specified.